### ISSUE TITLE
How to NOT load a category used in 'partition_on'?

### ISSUE THREAD

**Issue by yohplala** [dask/fastparquet#595]
Hi,

When loading a DataFrame with `to_pandas`, parameter `columns` allows restricting the list of columns to read.
But if a column has been used in `partition_on` parameter of `write`, then it is included in the columns of the returned DataFrame, even if not mentionned in parameter `columns`.

Please, is there any way to not include a column that is used in `partition_on`?

Example of code.
```python
from os import path as os_path
import fastparquet as fp
import pandas as pd

file = os_path.expanduser('~/Documents/code/data/test')
df = pd.DataFrame({'a':[1,2,3,4], 'b':['aha', 'oho', 'aha', 'oho']})
df['b'] = df['b'].astype('category')
fp.write(file, df, file_scheme='hive', partition_on=['b'])
df_rec = fp.ParquetFile(file).to_pandas(columns=['a'])
```

Result
```python
In [11]: df_rec
Out[11]: 
   a    b
0  1  aha
1  3  aha
2  2  oho
3  4  oho
```

**Comment by martindurant** [comment]
I believe you can do this if you load with dask (even with the fastparquet backend), but fastparquet doesn't have the feature, sorry.

**Comment by martindurant** [comment]
Fixed in #622 , coming soon

**Comment by yohplala** [comment]
Hi Martin, thanks to let me know. I will check that!
